PaymentInitiation/Program.cs

```diff
@@ -104,28 +114,29 @@ static async Task Main(string[] args)
             //
             // Get an API access token from auth server with the scope needed
             //
-            _token = await GetToken(_clientId, _clientSecret, _paymentinitiationScope);
+            _token = await GetToken(_settings.ClientId, _settings.ClientSecret, _paymentinitiationScope);
             Console.WriteLine($"token: {_token}");
             Console.WriteLine();
 
             //
             // Create the payment
             //
-            _payment.PaymentId = await CreatePaymentInitiation(_payment.BicFi, _payment.PaymentService, _payment.PaymentProduct, _payment.PaymentBody);
+            _payment.PaymentId = await CreatePaymentInitiation(_payment.BicFi, _settings.PSUIPAddress, _settings.PSUUserAgent, _payment.PSUCorporateId, _payment.PaymentService, _payment.PaymentProduct, _payment.PaymentBody, _payment.AffiliatedASPSPId);
             Console.WriteLine($"paymentId: {_payment.PaymentId}");
             Console.WriteLine();
 
             //
             // Create a payment authorization object to be used for authorizing the payment with the end user
             //
-            _payment.PaymentAuthId = await StartPaymentInitiationAuthorisationProcess(_payment.BicFi, _payment.PaymentService, _payment.PaymentProduct, _payment.PaymentId);
+            _payment.PaymentAuthId = await StartPaymentInitiationAuthorisationProcess(_payment.BicFi, _settings.PSUIPAddress, _settings.PSUUserAgent, _payment.PSUCorporateId, _payment.PaymentService, _payment.PaymentProduct, _payment.PaymentId);
             Console.WriteLine($"authId: {_payment.PaymentAuthId}");
             Console.WriteLine();
 
             //
             // Start the payment authorization process with the end user
             //
-            (_payment.ScaMethod, _payment.ScaData) = await UpdatePSUDataForPaymentInitiation(_payment.BicFi, _payment.PaymentService, _payment.PaymentProduct, _payment.PaymentId, _payment.PaymentAuthId);
+            string scaStatus = "";

```

Comment:
I don't think you need to initialize `scaStatus` here since you assign it on line 139